File: packages/dspwa1022/src/main/java/org/iu/dspwa1022/store/model/Customer.java
package org.iu.dspwa1022.store.model;

import java.util.ArrayList;
import java.util.List;
import java.util.UUID;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.Id;
import jakarta.persistence.OneToMany;
import jakarta.persistence.Table;

@Entity
@Table(schema = "dspwa1022", name = "customer")
public class Customer {

    @Id
    @GeneratedValue
    private UUID id;

    private String name;

    private String email;

    @OneToMany(mappedBy = "customer")
    private List<Order> orders = new ArrayList<>();

    public UUID getId() {
        return id;
    }

    public void setId(UUID id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

}

File: packages/dspwa1022-nestjs-demo/src/modules/store/entities/customer.ts
import { Column, Entity, PrimaryGeneratedColumn } from "typeorm";

@Entity({
  schema: "dspwa1022",
})
export class Customer {
  @PrimaryGeneratedColumn("uuid")
  id: string;

  @Column()
  name: string;

  @Column()
  email: string;

  @Column({
    name: "created_at",
  })
  createdAt: Date;
}
